src/builder.rs

```diff
@@ -1031,20 +996,14 @@ fn build_bundle<B, R: RngCore>(
                     .take(num_asset_pre_actions)
                     .collect::<Vec<_>>();
 
-                let mut indexed_outputs = outputs
+                let indexed_outputs = outputs
                     .into_iter()
                     .chain(iter::repeat_with(|| {
                         (OutputInfo::dummy(&mut rng, asset), None)
                     }))
                     .take(num_asset_pre_actions)
                     .collect::<Vec<_>>();
 
-                // Shuffle the spends and outputs, so that learning the position of a
-                // specific spent note or output note doesn't reveal anything on its own about
-                // the meaning of that note in the transaction context.
-                indexed_spends.shuffle(&mut rng);
-                indexed_outputs.shuffle(&mut rng);
-
```

Comment:
overall good, but i think we need to shuffle the spends and inputs anyway.

As discussed, separate the reproducible tests from the backward compatible implementation.